dbms/src/DataTypes/DataTypeDecimal.cpp

```diff
@@ -79,13 +80,50 @@ void DataTypeDecimal<T>::deserializeBinaryBulk(
     IColumn & column,
     ReadBuffer & istr,
     size_t limit,
-    double /*avg_value_size_hint*/) const
+    double /*avg_value_size_hint*/,
+    const IColumn::Filter * filter) const
```

Comment:
Add unit test about `deserializeBinaryBulk(..., filter)` to ensure the correctness for `DataTypeDecimal`/`DataTypeEnum`/`DataTypeNumberBase` and `DataTypeString`